gp-includes/formats/format-android.php

```diff
@@ -405,6 +405,21 @@ private function cmp_context( $a, $b ) {
 		return strnatcmp( $a->context, $b->context );
 	}
 
+	/**
+	 * Convert a string potentially containing escaped Unicode into UTF-8.
+	 *
+	 * @since 3.0
+	 *
+	 * @param string $string The string to unescape UTF-8.
+	 *
+	 * @return string Returns the unescaped string.
+	 */
+	private function unescape_unicode( $string ) {
+		return preg_replace_callback( '#\\\\u([0-9a-fA-F]{4})#', function ( $match ) {
+		    return mb_convert_encoding( pack( 'H*', $match[1] ), 'UTF-8', 'UTF-16BE' );
```

Comment:
Might it not be better to simply add in an additional slash so that stripcslashes() would then output the correct string?

This would preserve the original intent of the developer who wrote the string in the first place.

Reply:
Good idea, I've changed the code. Thanks!